controllers/platform/services/services.go

```diff
@@ -358,14 +371,26 @@ func (j JobServiceHandler) MergeContainerSpec(containerSpec *corev1.Container) (
 	return c, err
 }
 
+// hasPostgreSQLConfigured returns true when either the SonataFlow Platform PostgreSQL CR's structure or the one in the Job service specification is not nil
+func (j JobServiceHandler) hasPostgreSQLConfigured() bool {
+	return (j.platform.Spec.Services.JobService.Persistence != nil && j.platform.Spec.Services.JobService.Persistence.PostgreSQL != nil) ||
+		(j.platform.Spec.Persistence != nil && j.platform.Spec.Persistence.PostgreSQL != nil)
+}
+
 func (j JobServiceHandler) ConfigurePersistence(containerSpec *corev1.Container) *corev1.Container {
 
-	if j.platform.Spec.Services.JobService.Persistence != nil && j.platform.Spec.Services.JobService.Persistence.PostgreSql != nil {
+	if j.hasPostgreSQLConfigured() {
+		var p *operatorapi.PersistencePostgreSQL
 		c := containerSpec.DeepCopy()
 		c.Image = j.GetServiceImageName(constants.PersistenceTypePostgreSQL)
-		c.Env = append(c.Env, persistence.ConfigurePostgreSqlEnv(j.platform.Spec.Services.JobService.Persistence.PostgreSql, j.GetServiceName(), j.platform.Namespace)...)
+		if j.platform.Spec.Services.JobService.Persistence != nil && j.platform.Spec.Services.JobService.Persistence.PostgreSQL != nil {
```

Comment:
same

Reply:
Not sure because adding it to `hasPostgreSQLConfigured()` already validates it in line 382:
https://github.com/apache/incubator-kie-kogito-serverless-operator/pull/372/files/95b390b7469f45c3cfad661dd5d1f845757fcdc1#diff-9168d999a0cd028523626009a2534c70de62a6e4322770f8bc21d8fc01fc2b83R382

Reply:
assuming you add the proper checks to `hasPostgreSQLConfigured` then yes, agreed.

Reply:
Yes, it's hopefully on the last commit: https://github.com/apache/incubator-kie-kogito-serverless-operator/pull/372/commits/90af05ec683c1c6ac00dd4715314dbd3e265e4fd